modules/minimint-wallet/src/lib.rs

```diff
@@ -412,20 +410,72 @@ impl FederationModule for Wallet {
                     .map(|input| BatchItem::delete(UTXOKey(input.previous_output))),
             );
 
-            // Delete pe-outs from pending list
+            // Delete peg-outs from pending list
             batch.append_from_iter(
                 peg_out_ids
                     .into_iter()
                     .map(|peg_out| BatchItem::delete(PendingPegOutKey(peg_out))),
             );
 
-            batch.append_insert_new(UnsignedTransactionKey(psbt.global.unsigned_tx.txid()), psbt);
+            batch.append_insert_new(
+                UnsignedTransactionKey(psbt.global.unsigned_tx.txid()),
+                UnsignedTransaction {
+                    psbt,
+                    signatures: vec![],
+                },
+            );
             batch.append_insert_new(PegOutTxSignatureCI(txid), sigs);
         }
-        batch.commit();
 
-        // FIXME should use to drop non-contributing peers
-        vec![]
+        // Sign and finalize any unsigned transactions that have signatures
+        let unsigned_txs: Vec<(UnsignedTransactionKey, UnsignedTransaction)> = self
+            .db
+            .find_by_prefix(&UnsignedTransactionPrefixKey)
+            .map(|res| res.expect("DB error"))
+            .filter(|(_, unsigned)| !unsigned.signatures.is_empty())
+            .collect();
+
+        let mut drop_peers = Vec::<PeerId>::new();
+        for (key, unsigned) in unsigned_txs {
+            let UnsignedTransaction {
+                mut psbt,
+                signatures,
+            } = unsigned;
+
+            let signers: HashSet<PeerId> = signatures
+                .iter()
+                .filter_map(
+                    |(peer, sig)| match self.sign_peg_out_psbt(&mut psbt, peer, sig) {
+                        Ok(_) => Some(*peer),
+                        Err(error) => {
+                            warn!("Error with {} partial sig {:?}", peer, error);
+                            None
+                        }
+                    },
+                )
+                .collect();
+
+            for peer in consensus_peers.sub(&signers) {
+                error!("Dropping {:?} for not contributing sigs to PSBT", peer);
+                drop_peers.push(peer);
+            }
```

Comment:
nit: this could probably use `Vec::extend` instead (with `map` for error reporting in between) which could make use of the fact that `<HashSet as IntoIterator>::IntoIter: ExactSizeIterator` and thus be slightly more efficient. But we don't intend to drop a lot of peers, so this point is moot.

Reply:
Considering this should almost never happen, I may leave this because I think it reads a little easier.